patches/api/0384-More-Teleport-API.patch

```diff
@@ -0,0 +1,282 @@
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/LookAnchor.java
+@@ -0,0 +1,24 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.Location;
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.LivingEntity;
++
++/**
++ * Represents what part of the entity should be used when determining where to face a location/entity.
++ *
++ * @see org.bukkit.entity.Player#lookAt(Location, LookAnchor)
++ * @see org.bukkit.entity.Player#lookAt(Entity, LookAnchor, LookAnchor)
++ */
++public enum LookAnchor {
++    /**
++     * Represents the entity's feet.
++     * @see LivingEntity#getLocation()
++     */
++    FEET,
++    /**
++     * Represents the entity's eyes.
++     * @see LivingEntity#getEyeLocation()
++     */
++    EYES;
++}
+diff --git a/src/main/java/io/papermc/paper/entity/RelativeTeleportFlag.java b/src/main/java/io/papermc/paper/entity/RelativeTeleportFlag.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..91dcc8a4ca6d2179fe9ffb079896c7722eae2117
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/entity/RelativeTeleportFlag.java
+@@ -0,0 +1,33 @@
++package io.papermc.paper.entity;
++
++import org.bukkit.Location;
++import org.bukkit.event.player.PlayerTeleportEvent;
++
++/**
++ * Represents coordinates in a teleportation that should be handled relatively.
++ *
++ * @see org.bukkit.entity.Player#teleport(Location, PlayerTeleportEvent.TeleportCause, boolean, boolean, RelativeTeleportFlag...)
++ */
++public enum RelativeTeleportFlag {
++    /**
++     * Represents the player's X coordinate
++     */
++    X,
++    /**
++     * Represents the player's Y coordinate
++     */
++    Y,
++    /**
++     * Represents the player's Z coordinate
++     */
++    Z,
++    /**
++     * Represents the player's yaw
++     */
++    YAW,
++    /**
++     * Represents the player's pitch
++     */
++    PITCH;
++
++}
+diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
+index 8bc6876c82935988436597161fa0ec94c032174b..1e3bc6de9529f7ef4e1ff47f43a803b7a6af856f 100644
+--- a/src/main/java/org/bukkit/entity/Entity.java
++++ b/src/main/java/org/bukkit/entity/Entity.java
+@@ -121,10 +121,73 @@ public interface Entity extends Metadatable, CommandSender, Nameable, Persistent
+      *
+      * @param yaw the yaw
+      * @param pitch the pitch
+-     * @throws UnsupportedOperationException if used for players
+      */
+     public void setRotation(float yaw, float pitch);
+ 
++    // Paper start - Teleport API
++    /**
++     * Teleports this entity to the given location.
++     * <p>
++     * Note: Teleporting to a different world with ignorePassengers to true while the entity has entities riding it
++     * will cause this teleportation to return false and not occur.
++     *
++     * @param location New location to teleport this entity to
++     * @param ignorePassengers If all passengers should be required to be removed prior to teleportation
```

Comment:
I interpret this as reversed now. I get the impression that setting ignorePassengers to true would require removing the passengers, but it's the other way around.